contributions/api/controllers/contributions.py

```diff
@@ -125,36 +125,57 @@ def post():
 def search():
     args = request.args
     query = dict()
-    try:
-        query = query_params.format_query_contribution(args, query)
-    except Exception as ex:
-        msg = {
-            "reason": "The query is wrong or bad argument " + str(args),
-            "error": "Bad Request: " + request.url,
-        }
-        msg_json = jsonutils.create_log_json("Contribution", "SEARCH", msg)
-        logging.error("Contribution SEARCH " + json.dumps(msg_json))
-        return rs_handlers.bad_request(msg_json)
+    if len(args) > 0:   # it is search
```

Comment:
I think it might be possible to simplify the current implementation. When query is `None` (currently set here `query = query_params.format_query_contribution(args, query)` when there are no args ), PyMongo `find` methods return all documents in the collection. So, we may not additional util method and condition to check length of arguments here? What do you think?

Reply:
I tried that way first but it got failed and I don't remember exactly why, maybe I did something wrong, so I made this way. But if you want to change that way, I will try to find a way to make that way. Let me know if you want to change

Reply:
I found out why it didn't work before. I will update it

Reply:
I think that the code will be much more complicated if we do that... Maybe current is probably the cleanest.

Reply:
If for say `GET /contributions?name1=test` or other unsupported query parameters, we can return HTTP 400 Bad Request, would we be able to pass None to PyMongo method if there are no query parameters?

Reply:
So you mean, if it is bad query, give 400 instead of empty list [] ? What do you mean by `would we be able to pass None to PyMongo method if there are no query parameters?` Did you mean that it gives query values as none to pymongo?

Reply:
Yes, if it is bad/unrecognized query parameters, let's return 400 instead of empty list. That seems to be the recommended approach. Based on that, I was wondering, if that would help in simplifying the current implementation (first comment in this conversation). If they are unrelated, then we can proceed with the current implementation.

Reply:
I believe so. I will try to make it to return 400 if it is not a correct query

Reply:
Updated to give 400 error when using a bad keyword for search